src/util-inl.h

```diff
@@ -542,6 +543,23 @@ void ArrayBufferViewContents<T, S>::Read(v8::Local<v8::ArrayBufferView> abv) {
   }
 }
 
+template <typename T, size_t S>
+void ArrayBufferViewContents<T, S>::ReadValue(v8::Local<v8::Value> buf) {
+  static_assert(sizeof(T) == 1, "Only supports one-byte data at the moment");
+
+  if (buf->IsArrayBufferView()) {
+    Read(buf.As<v8::ArrayBufferView>());
+  } else if (buf->IsArrayBuffer()) {
+    auto ab = buf.As<v8::ArrayBuffer>();
+    length_ = ab->ByteLength();
+    data_ = reinterpret_cast<T*>(ab->Data());
+  } else {
+    auto sab = buf.As<v8::SharedArrayBuffer>();
+    length_ = sab->ByteLength();
+    data_ = reinterpret_cast<T*>(sab->Data());
+  }
```

Comment:
What happens if `buf` is neither `ArrayBufferView` nor `ArrayBuffer` nor `SharedArrayBuffer`?

Reply:
Ah, I see. I'll also add the check here. Good catch. Thank you for the review!